pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -315,6 +315,11 @@ func (d *DeploymentController) configureIstioGateway(log *istiolog.Scope, gw gat
 	}
 	for _, t := range rendered {
 		if err := d.apply(gi.controller, t); err != nil {
+			log.Warn("Gateway deployment controller failed to apply template",
```

Comment:
any error in this function is already logged so logging an error here seems a bit odd. Note this will also dump like 1k lines of YAML... maybe at debug level?

Reply:
Yes, I can move it to info. The error is mostly useless - "error on line xxx" - I don't think anyone will be able to figure out what is wrong without the 1k yaml.

Reply:
Yes but if there is another error like permission denied or conflict - which can happen routinely - they will not be happy to see 1k lines logged, with a tight retry loop

Reply:
Something has to give... Can we log it once per something ? 

My concern is that it was simply not possible ( at least for me ) to figure out what is wrong without adding this line.
( or run istio locally with a breakpoint and inspect - but that's even worse ).